core/src/main/java/io/micronaut/core/type/Argument.java

```diff
@@ -174,6 +174,37 @@ default boolean isAssignableFrom(@NonNull Class<?> candidateType) {
         return getType().isAssignableFrom(Objects.requireNonNull(candidateType, "Candidate type cannot be null"));
     }
 
+    /**
+     * Checks if the argument can be assigned to this argument.
+     * @param candidateArgument The candidate argument
+     * @return True if it is assignable from.
+     * @since 3.0.0
+     */
+    default boolean isAssignableFrom(@NonNull Argument candidateArgument) {
+        Objects.requireNonNull(candidateArgument, "Candidate type cannot be null");
+        if (!isAssignableFrom(candidateArgument.getType())) {
+            return false;
+        }
+        Argument[] typeParameters = getTypeParameters();
+        Argument[] candidateArgumentTypeParameters = candidateArgument.getTypeParameters();
+        if (typeParameters.length == 0) {
+            // Wildcard or no type parameters
+            return candidateArgumentTypeParameters.length >= 0;
+        }
+        if (candidateArgumentTypeParameters.length == 0) {
+            // Wildcard
+            return false;
+        }
+        for (int i = 0; i < typeParameters.length; i++) {
+            Argument typeParameter = typeParameters[i];
+            Argument candidateArgumentTypeParameter = candidateArgumentTypeParameters[i];
+            if (!typeParameter.isAssignableFrom(candidateArgumentTypeParameter)) {
```

Comment:
Do we need to check `isTypeVariable`?

Reply:
We know it's coming from the type parameters so I don't think so